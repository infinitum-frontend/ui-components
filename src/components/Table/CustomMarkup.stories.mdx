import { Meta, Unstyled, ArgTypes } from '@storybook/blocks'
import Table from './Table'
import TableBase from 'Components/Table/components/TableBase'

<Meta title="Components/Table/CustomMarkup" />

## Кастомный вариант таблицы

Если в таблицу передается проп `children`, появляется возможность создать легковесную таблицу произвольного формата.
При этом функционал, связанный с сортировкой, фильтрацией, селекцией, изменением размера становится недоступным, а также отпадает необходимость создавать объекты columns и data и передавать их в таблицу.

<Unstyled>
  <Table>
    <thead>
      <tr>
        <th>Заголовок 1</th>
        <th>Заголовок 2</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>Ячейка 1</td>
        <td>Ячейка 2</td>
      </tr>
    </tbody>
  </Table>
</Unstyled>

{/* prettier-ignore-start */}
```html
<Table>
  <thead>
    <tr>
      <th>Заголовок 1</th>
      <th>Заголовок 2</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Ячейка 1</td>
      <td>Ячейка 2</td>
    </tr>
  </tbody>
</Table>
```
{/* prettier-ignore-end */}

<ArgTypes of={TableBase} />
