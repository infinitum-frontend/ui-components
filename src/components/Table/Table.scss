.inf-table {
  border-collapse: separate;
  border-spacing: 0;
  width: 100%;
  font-family: var(--inf-font-family-base);
  --table-border-radius: 0;
  border-radius: var(--table-border-radius);

  thead {
    th {
      background-color: var(--inf-color-background-tertiary);
      text-align: left;
      color: var(--inf-color-text-secondary);
      font: var(--inf-font-body-1);
      padding: var(--inf-space-medium);
      position: relative;
      border-top: 1px solid var(--inf-color-border-primary);

      &:hover .inf-table-header__resizer {
        opacity: 1;
      }

      &:first-child {
        border-top-left-radius: var(--table-border-radius);
        border-left: 1px solid var(--inf-color-border-primary);
      }

      &:last-child {
        border-top-right-radius: var(--table-border-radius);
        border-right: 1px solid var(--inf-color-border-primary);
      }
    }
  }

  tbody {
    tr {
      font: var(--inf-font-body-2);

      @include hover {
        background-color: var(--inf-color-background-primary-hover);
      }

      &.inf-table__row--interactive {
        cursor: pointer;
      }

      &.inf-table__row--selected {
        background-color: var(
          --inf-color-background-primary-hover
        ); // TODO: уточнить цвет
      }

      &:not(:last-child) {
        border-bottom: 1px solid var(--inf-color-border-primary);
      }
    }

    tr:last-child td:first-child {
      border-bottom-left-radius: var(--table-border-radius);
    }

    tr:last-child td {
      border-bottom: 1px solid var(--inf-color-border-primary);
    }

    tr:last-child td:last-child {
      border-bottom-right-radius: var(--table-border-radius);
    }

    td {
      padding: var(--inf-space-medium);
      font: var(--inf-font-body-1);
      color: var(--inf-color-text-primary);
      border-top: 1px solid var(--inf-color-border-primary);
    }

    td:first-child {
      border-left: 1px solid var(--inf-color-border-primary);
    }

    td:last-child {
      border-right: 1px solid var(--inf-color-border-primary);
    }
  }

  &--border-radius-xsmall {
    --table-border-radius: var(--inf-border-radius-xsmall);
  }

  &--border-radius-small {
    --table-border-radius: var(--inf-border-radius-small);
  }

  &--border-radius-medium {
    --table-border-radius: var(--inf-border-radius-medium);
  }

  &--border-radius-large {
    --table-border-radius: var(--inf-border-radius-large);
  }

  &--borderless {
    --table-border-radius: 0;

    thead th {
      border-top: none;

      &:first-child,
      &:last-child {
        border-left: none;
        border-right: none;
      }
    }

    tbody {
      tr td {
        &:first-child,
        &:last-child {
          border-left: none;
          border-right: none;
        }
      }

      tr:last-child td {
        border-bottom: none;
      }

      tr:first-child td {
        border-top: none;
      }
    }
  }

  $vertical-align-values: (
    'baseline',
    'bottom',
    'middle',
    'sub',
    'super',
    'text-bottom',
    'text-top',
    'top'
  );

  @mixin vertical-align($vertical-align-value) {
    &--vertical-align-#{$vertical-align-value} td {
      vertical-align: #{$vertical-align-value};
    }
  }

  @each $vertical-align-value in $vertical-align-values {
    @include vertical-align($vertical-align-value);
  }
}
