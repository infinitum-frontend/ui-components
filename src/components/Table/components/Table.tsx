import {
  CSSProperties,
  ReactElement,
  TableHTMLAttributes,
  useState
} from 'react'
import {
  ColumnDef,
  getCoreRowModel,
  getFilteredRowModel,
  useReactTable,
  SortingState,
  getSortedRowModel
} from '@tanstack/react-table'
import cn from 'classnames'
import TableHeader from 'Components/Table/components/Header'
import TableBody from 'Components/Table/components/Body'
import '../index.scss'

interface BaseRow {
  className?: string
  style?: CSSProperties
}

export type Row<T extends Record<any, any>> = BaseRow & T
export interface TableProps extends TableHTMLAttributes<HTMLTableElement> {
  /** Массив с данными для построения шапки таблицы */
  columns: Array<ColumnDef<any, any>>
  /** Массив с данными для построения тела таблицы */
  rows: Array<Row<any>>
  /** Максимальное количество отображаемых элементов */
  maxLength?: number
  /** Включение сортировки по столбцам */
  withSorting?: boolean
  sortingState?: SortingState
  // /** Отображение чекбоксов в 1 колонке */
  // withRowSelection?: boolean
  // withFiltering?: boolean
  // /** Событие изменения чекбоксов */
  // onChangeRowSelection?: OnChangeFn<RowSelectionState>
  // /** Обьект с состоянием чекбоксов */
  // selectionValue?: RowSelectionState
  // onFiltersChange?: (state: ColumnFiltersState) => void
  // /** Начальное состояние фильтров */
  // filtersState?: ColumnFiltersState
  // /**
  //  * Режим фильтрации
  //  * @default auto
  //  * auto - таблица сама фильтрует среди всех своих данных,
  //  * manual - фильтры не устанавливаются, а только вызывается событие onFiltersChange с состоянием фильтров
  //  */
  // // filterMode?: 'auto' | 'manual'
  // /** Включена ли группировка */
  // // enableGrouping?: boolean
}

const Table = ({
  columns = [],
  rows,
  className,
  maxLength = 5,
  // withRowSelection,
  withSorting,
  // withFiltering,
  // onChangeRowSelection,
  // selectionValue = {},
  // onFiltersChange,
  // filtersState = [],
  // enableGrouping = false,
  // filterMode = 'manual',
  // sortingMode = 'manual
  ...props
}: TableProps): ReactElement => {
  // const filterMode = 'manual'
  // const [rowSelection, setRowSelection] = useState(selectionValue)
  const [sorting, setSorting] = useState<SortingState>([])
  // const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>(
  //   filtersState || []
  // )
  // const [manualFilters, setManualFilters] = useState<ColumnFiltersState>(
  //   filtersState || []
  // )

  // const memoizedColumns = useMemo(() => {
  //   if (withRowSelection) {
  //     const deepCopy: Array<ColumnDef<any>> = JSON.parse(
  //       JSON.stringify(columns)
  //     )
  //     deepCopy.unshift({
  //       id: 'checkbox',
  //       header: ({ table }) => {
  //         return (
  //           <Checkbox
  //             checked={Boolean(table.getIsAllRowsSelected())}
  //             indeterminate={table.getIsSomeRowsSelected()}
  //             onChange={(value, e) =>
  //               table.getToggleAllRowsSelectedHandler().call({}, e)
  //             }
  //           />
  //         )
  //       },
  //       cell: ({ row }) => {
  //         return (
  //           <div className="px-1">
  //             <Checkbox
  //               checked={row.getIsSelected()}
  //               indeterminate={row.getIsSomeSelected()}
  //               onChange={row.getToggleSelectedHandler()}
  //             />
  //           </div>
  //         )
  //       }
  //     })
  //
  //     return deepCopy
  //   }
  //
  //   return columns
  // }, [...columns, withRowSelection])

  // const handleRowSelection: OnChangeFn<RowSelectionState> = (callback) => {
  //   const newState =
  //     typeof callback === 'function' ? callback(rowSelection) : {}
  //   setRowSelection(newState)
  //   onChangeRowSelection?.(newState)
  // }
  //
  // const handleFiltersChange: (value: string, column: Column<any>) => void = (
  //   value,
  //   column
  // ) => {
  // if (filterMode === 'auto') {
  //   column.setFilterValue(value)
  // }

  //   if (filterMode === 'manual') {
  //     const newState = [
  //       ...manualFilters?.filter((item) => item.id !== column.id),
  //       value ? { id: column.id, value } : undefined
  //     ].filter((item) => Boolean(item)) as ColumnFiltersState
  //     setManualFilters(newState)
  //     onFiltersChange?.(newState)
  //   }
  // }

  const table = useReactTable({
    data: rows,
    columns,
    getFilteredRowModel: getFilteredRowModel(),
    getCoreRowModel: getCoreRowModel(),
    state: {
      // rowSelection,
      // columnFilters,
      sorting
    },
    // filterFns: {
    //   elIncludesString: (row, columnId, filterValue) => {
    //     return getByText(row, columnId, filterValue)
    //   }
    // },
    // onColumnFiltersChange: setColumnFilters,
    onSortingChange: setSorting,
    // getSubRows: (row) => row?.subRows,
    // manualGrouping: enableGrouping,
    // onRowSelectionChange: handleRowSelection,
    getSortedRowModel: getSortedRowModel()
    // getFacetedRowModel: getFacetedRowModel(),
    // getFacetedUniqueValues: getFacetedUniqueValues(),
    // getFacetedMinMaxValues: getFacetedMinMaxValues()
  })

  const tableRows = table.getRowModel().rows.slice(0, maxLength)

  return (
    <table className={cn('inf-table', className)} {...props}>
      <TableHeader
        table={table}
        withSorting={withSorting}
        // withFiltering={withFiltering}
        // onFiltersChange={handleFiltersChange}
      />
      <TableBody
        rows={tableRows}
        // grouping={enableGrouping}
      />
    </table>
  )
}

export default Table
