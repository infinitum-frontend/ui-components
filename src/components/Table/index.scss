.inf-table {
  border-collapse: collapse;
  width: 100%;
  font-family: var(--inf-font-family-base);

  th {
    text-align: left;
    color: var(--inf-color-text-secondary);
    font-weight: var(--inf-font-weight-normal);
    font-size: var(--inf-text-size-small);
    padding: 6px;
  }

  tbody {
    tr {
      &:hover {
        background-color: var(--inf-control-background-color-hover);
      }
      &:first-child td {
        padding-top: 12px;
      }

      td {
        padding: 6px;
      }

      &.inf-table__group-label {
        background-color: #F2F4F5;
        text-transform: uppercase;

        td {
          padding: 0 6px;
        }
      }
    }
  }
}
