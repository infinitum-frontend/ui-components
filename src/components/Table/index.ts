// Сначала экспортируем наши типы, чтобы они были в приоритете над танстаковскими
export type { OnChangeFn } from 'Utils/types'
export * from './types'
export * from '@tanstack/react-table'

export { default as Table } from './Table'
