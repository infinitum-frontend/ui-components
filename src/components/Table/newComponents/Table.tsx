import { ComponentPropsWithoutRef, ReactElement } from 'react'
import TableBody from './TableBody'
import TableCell from './TableCell'
import TableFilterPopover from './TableFilterPopover'
import TableHeader from './TableHeader'
import TableHeaderCell from './TableHeaderCell'
import TableHeaderRow from './TableHeaderRow'
import TableRow from './TableRow'
import TableSort from './TableSort'

export interface TableProps extends ComponentPropsWithoutRef<'table'> {}

const Table = ({ children, ...restProps }: TableProps): ReactElement => {
  return (
    <table className="inf-new-table" {...restProps}>
      {children}
    </table>
  )
}

export default Object.assign(Table, {
  Header: TableHeader,
  HeaderCell: TableHeaderCell,
  HeaderRow: TableHeaderRow,
  Body: TableBody,
  Cell: TableCell,
  Row: TableRow,
  FilterPopover: TableFilterPopover,
  Sort: TableSort
})
