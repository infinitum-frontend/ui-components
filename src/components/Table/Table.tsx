// eslint-disable-next-line @typescript-eslint/no-unused-vars
import React, { ReactElement, useEffect, useMemo } from 'react'
import {
  Column,
  ColumnDef,
  ColumnMeta,
  ColumnResizeMode,
  getCoreRowModel,
  getFacetedUniqueValues,
  OnChangeFn as TanstackOnChangeFn,
  RowSelectionState,
  SortingState,
  useReactTable
} from '@tanstack/react-table'
import cn from 'classnames'
import TableHeader from './components/TableHeader'
import TableBody from './components/TableBody'
import { Checkbox } from 'Components/Checkbox'
import {
  TableColumnFiltersState,
  TableColumnFilterValue,
  TableRow,
  TableRowData,
  TableSelectedRow,
  TableSelectionState,
  TableVerticalAlignValue
} from './types'
import { OnChangeFn } from 'Utils/types'
import { mapRowToExternalFormat } from './helpers'
import TableBase, { TableBaseProps } from './components/TableBase'
import './Table.scss'

export interface TableProps extends TableBaseProps {
  /** Массив с данными для построения шапки таблицы */
  columns?: Array<ColumnDef<any, any>>
  /** Массив с данными для построения тела таблицы */
  rows?: Array<TableRowData<any>>
  /** Скругление границ таблицы */
  borderRadius?: 'xsmall' | 'small' | 'medium' | 'large'
  /** CSS свойство vertical-align для шапки */
  verticalAlignHead?: TableVerticalAlignValue
  /** CSS свойство vertical-align для рядов */
  verticalAlignBody?: TableVerticalAlignValue
  /**
   * @deprecated
   * Максимальное количество отображаемых элементов
   */
  maxLength?: number
  /** Включение сортировки по столбцам */
  withSorting?: boolean
  /** Начальное состояние сортировки */
  sortingState?: SortingState
  /** Событие изменения состояния сортировки */
  onSortingChange?: OnChangeFn<SortingState>
  // TODO: sorting mode auto
  /** @deprecated */
  withFiltering?: boolean
  /**
   * @deprecated
   * Событие изменения состояния фильтров
   */
  onFiltersChange?: OnChangeFn<TableColumnFiltersState>
  /**
   * @deprecated
   * Начальное состояние фильтров
   */
  filtersState?: TableColumnFiltersState
  /** Отображение чекбоксов в 1 колонке */
  withRowSelection?: boolean
  /** Событие изменения чекбоксов */
  onChangeRowSelection?: OnChangeFn<TableSelectionState<any>>
  /** Состояние выбора рядов через чекбокс */
  selectionState?: TableSelectionState<any>
  /**
   * Выбранный ряд. Если передается строка или число, идет сравнение аргумента с id ряда.
   */
  selectedRow?: TableSelectedRow
  onRowClick?: OnChangeFn<TableRow>
  /** Изменение ширины колонок
   * @value onChange изменение "вживую" при растягивании
   * @value onEnd изменение при отжатии кнопки мыши
   */
  resizeMode?: ColumnResizeMode
  /** видимость колонок
   * в качестве данных передается объект с ключами, соответствующими id колонок,
   * имеющим булевы значения, отражающими видимость колонки
   */
  columnVisibility?: Record<string, boolean>
  // /** Включена ли группировка */
  // // enableGrouping?: boolean
}

/** Компонент многофункциональной таблицы */
const Table = ({
  columns = [],
  rows = [],
  className,
  borderRadius,
  verticalAlignHead = 'top',
  verticalAlignBody,
  maxLength,
  withSorting,
  onSortingChange,
  sortingState = [],
  withFiltering,
  onFiltersChange,
  filtersState = [],
  withRowSelection,
  onChangeRowSelection,
  selectionState = [],
  selectedRow,
  onRowClick,
  resizeMode,
  columnVisibility = {},
  // enableGrouping = false,
  children,
  ...props
}: TableProps): ReactElement => {
  if (children) {
    return (
      <TableBase
        verticalAlignBody={verticalAlignBody}
        verticalAlignHead={verticalAlignHead}
        borderRadius={borderRadius}
        {...props}
      >
        {children}
      </TableBase>
    )
  }

  // Приводим состояние селекции столбцом к формату танстака(объект ключ-id ряда, значение-boolean)
  const tanstackSelectionState = useMemo(
    () =>
      selectionState?.reduce<RowSelectionState>((accumulator, currentValue) => {
        accumulator[currentValue.id] = true
        return accumulator
      }, {}),
    [selectionState]
  )

  // ==================== handlers ====================
  const handleSortingChange: (state: SortingState) => void = (state) => {
    onSortingChange?.(state)
  }

  const handleFiltersChange: (
    value: TableColumnFilterValue,
    filterType: ColumnMeta<any, any>['filterType'],
    column: Column<any>
  ) => void = (value, filterType, column) => {
    const newState = [
      ...filtersState?.filter((item) => item.id !== column.id),
      value ? { id: column.id, filterType, value } : undefined
    ].filter((item) => Boolean(item)) as TableColumnFiltersState
    onFiltersChange?.(newState)
  }

  // маппим данные ряда из формата танстака к нашему формату
  const handleRowSelection: TanstackOnChangeFn<RowSelectionState> = (
    callback
  ) => {
    const newTanstackSelectionState =
      typeof callback === 'function' ? callback(tanstackSelectionState) : {}

    const rowSelectionState: Array<TableRow<any>> = []
    Object.keys(newTanstackSelectionState).forEach((key) => {
      const row = mapRowToExternalFormat(table.getRow(key))
      rowSelectionState.push(row)
    })

    onChangeRowSelection?.(rowSelectionState)
  }

  const memoizedColumns = useMemo(() => {
    if (withRowSelection) {
      return [
        {
          id: 'checkbox',
          size: 20,
          header: ({ table }) => {
            return (
              <Checkbox
                checked={Boolean(table.getIsAllRowsSelected())}
                indeterminate={table.getIsSomeRowsSelected()}
                onClick={(e) => e.stopPropagation()}
                onChange={(value, e) =>
                  table.getToggleAllRowsSelectedHandler().call({}, e)
                }
              />
            )
          },
          cell: ({ row }) => {
            return (
              <div>
                <Checkbox
                  checked={row.getIsSelected()}
                  indeterminate={row.getIsSomeSelected()}
                  onClick={(e) => e.stopPropagation()}
                  onChange={row.getToggleSelectedHandler()}
                />
              </div>
            )
          }
        },
        ...columns
      ] as Array<ColumnDef<any>>
    }

    return columns
  }, [columns, withRowSelection])

  const table = useReactTable({
    data: rows,
    columns: memoizedColumns,
    columnResizeMode: resizeMode,
    getCoreRowModel: getCoreRowModel(),
    state: {
      rowSelection: tanstackSelectionState,
      sorting: sortingState,
      columnVisibility
    },
    manualFiltering: true,
    // manualGrouping: enableGrouping,
    onRowSelectionChange: handleRowSelection,
    getFacetedUniqueValues: getFacetedUniqueValues()
    // getSubRows: (row) => row?.subRows,
  })

  const tableRows = maxLength
    ? table.getRowModel().rows.slice(0, maxLength)
    : table.getRowModel().rows

  return (
    <table
      className={cn('inf-table', className, {
        [`inf-table--border-radius-${borderRadius as string}`]: borderRadius,
        'inf-table--bordered': props.bordered
      })}
      {...props}
    >
      <TableHeader
        table={table}
        withSorting={withSorting}
        withFiltering={withFiltering}
        sortingState={sortingState}
        onSortingChange={handleSortingChange}
        filtersState={filtersState}
        onFiltersChange={handleFiltersChange}
        resizeMode={resizeMode}
        verticalAlignHead={verticalAlignHead}
      />
      <TableBody
        rows={tableRows}
        selectedRow={selectedRow}
        onRowClick={onRowClick}
        // grouping={enableGrouping}
        verticalAlignBody={verticalAlignBody}
      />
    </table>
  )
}

export default Table
