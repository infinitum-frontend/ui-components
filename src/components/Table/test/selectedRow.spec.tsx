import { renderComponent } from '~/testSetup'
import { TABLE_COLUMNS, TABLE_DATA } from '../fixtures'
import Table from '../Table'
import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

// Тест кейсы
// если onRowClick указан, ряд при наведении имеет класс inf-table-row--interactive +
// если selectedRow задан, указанный ряд имеет соответствующий бэкграунд +
// selectedRow как функция (также добавить тест с поддержкой getRowId из IDD-593) +
// onRowClick корректно обрабатывается +

describe('Table selected row', () => {
  it('should match snapshot', () => {
    const { el } = renderComponent(
      <Table
        rows={TABLE_DATA}
        columns={TABLE_COLUMNS}
        selectedRow="0"
        onRowClick={() => {}}
      />
    )

    expect(el).toMatchSnapshot()
  })

  it('should apply className', () => {
    renderComponent(
      <Table
        rows={TABLE_DATA}
        columns={TABLE_COLUMNS}
        selectedRow="0"
        onRowClick={() => {}}
      />
    )

    const rows = screen.queryAllByRole('row')
    expect(rows[1].className).toMatch('inf-table-row--selected')

    rows.forEach((row, index) => {
      if (index !== 0) {
        expect(row.className).toMatch('inf-table-row--interactive')
      }
    })
  })

  it('should use custom function for selected row', () => {
    const selectedPortfolioName = TABLE_DATA[2].portfolio

    renderComponent(
      <Table
        rows={TABLE_DATA}
        columns={TABLE_COLUMNS}
        selectedRow={(row) => row.rowData.portfolio === selectedPortfolioName}
        onRowClick={() => {}}
      />
    )

    const rows = screen.queryAllByRole('row')

    expect(rows[2 + 1].className).toMatch('inf-table-row--selected')
  })

  it('should handle on row click correctly', async () => {
    const selectedPortfolio = TABLE_DATA[2]

    const rowClickHandler = vi.fn()

    renderComponent(
      <Table
        rows={TABLE_DATA}
        columns={TABLE_COLUMNS}
        selectedRow={undefined}
        onRowClick={rowClickHandler}
      />
    )

    const rows = screen.queryAllByRole('row')

    const user = userEvent.setup()

    // Поскольку клик по факту происходит на td, обрабатываем его
    await user.click(rows[3].children[0])

    expect(rowClickHandler).toBeCalledTimes(1)

    expect(rowClickHandler).toBeCalledWith({
      id: '2',
      rowData: selectedPortfolio
    })
  })
})
