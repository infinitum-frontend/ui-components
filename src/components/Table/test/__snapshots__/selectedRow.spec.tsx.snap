// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Table selected row > should match snapshot 1`] = `
<table
  class="inf-table"
>
  <thead
    class="inf-table-header"
  >
    <tr
      class="inf-table-header-row"
    >
      <th
        class="inf-table-header-cell"
        colspan="1"
        style="width: 150px;"
      >
        <div
          class="inf-table-header-cell__content"
        >
          <div
            class="inf-table-header-cell__heading"
            data-selector="heading"
            data-tooltip-state="closed"
          >
            Портфель
          </div>
          <div
            class="inf-table-header-cell__buttons"
          />
        </div>
      </th>
      <th
        class="inf-table-header-cell"
        colspan="1"
        style="width: 150px;"
      >
        <div
          class="inf-table-header-cell__content"
        >
          <div
            class="inf-table-header-cell__heading"
            data-selector="heading"
            data-tooltip-state="closed"
          >
            Показатель
          </div>
          <div
            class="inf-table-header-cell__buttons"
          />
        </div>
      </th>
      <th
        class="inf-table-header-cell"
        colspan="1"
        style="width: 150px;"
      >
        <div
          class="inf-table-header-cell__content"
        >
          <div
            class="inf-table-header-cell__heading"
            data-selector="heading"
            data-tooltip-state="closed"
          >
            Тип
          </div>
          <div
            class="inf-table-header-cell__buttons"
          />
        </div>
      </th>
      <th
        class="inf-table-header-cell"
        colspan="1"
        style="width: 150px;"
      >
        <div
          class="inf-table-header-cell__content"
        >
          <div
            class="inf-table-header-cell__heading"
            data-selector="heading"
            data-tooltip-state="closed"
          >
            Статус
          </div>
          <div
            class="inf-table-header-cell__buttons"
          />
        </div>
      </th>
      <th
        class="inf-table-header-cell"
        colspan="1"
        style="width: 150px;"
      >
        <div
          class="inf-table-header-cell__content"
        >
          <div
            class="inf-table-header-cell__heading"
            data-selector="heading"
            data-tooltip-state="closed"
          >
            Дата
          </div>
          <div
            class="inf-table-header-cell__buttons"
          />
        </div>
      </th>
    </tr>
  </thead>
  <tbody
    class="inf-table-body"
  >
    <tr
      class="inf-table-row inf-table-row--selected inf-table-row--interactive inf-table-row--hoverable"
    >
      <td
        class="inf-table-cell"
      >
        ОПИФ Открытие
      </td>
      <td
        class="inf-table-cell"
      >
        2335
      </td>
      <td
        class="inf-table-cell"
      >
        УВНР
      </td>
      <td
        class="inf-table-cell"
      >
        Сформировано
      </td>
      <td
        class="inf-table-cell"
      >
        2024-10-31
      </td>
    </tr>
    <tr
      class="inf-table-row inf-table-row--interactive inf-table-row--hoverable"
    >
      <td
        class="inf-table-cell"
      >
        ЗПИФ И Открытие — ИММВБ — высокая капитализация ОПИФ И Открытие — ИММВБ — высокая капитализация ОПИФ И Открытие — ИММВБ — высокая капитализация ОПИФ И Открытие — ИММВБ — высокая капитализация
      </td>
      <td
        class="inf-table-cell"
      >
        2206
      </td>
      <td
        class="inf-table-cell"
      >
        УВУН
      </td>
      <td
        class="inf-table-cell"
      >
        Сформировано
      </td>
      <td
        class="inf-table-cell"
      />
    </tr>
    <tr
      class="inf-table-row inf-table-row--interactive inf-table-row--hoverable"
    >
      <td
        class="inf-table-cell"
      >
        Закрытие — Telecommunicaton Index
      </td>
      <td
        class="inf-table-cell"
      >
        2206
      </td>
      <td
        class="inf-table-cell"
      >
        УВУН
      </td>
      <td
        class="inf-table-cell"
      >
        Сформировано
      </td>
      <td
        class="inf-table-cell"
      >
        2020-01-01
      </td>
    </tr>
    <tr
      class="inf-table-row inf-table-row--interactive inf-table-row--hoverable"
    >
      <td
        class="inf-table-cell"
      >
        ЯРА И Открытие — ИММВБ — машиностроение
      </td>
      <td
        class="inf-table-cell"
      >
        2206
      </td>
      <td
        class="inf-table-cell"
      >
        УВНУ
      </td>
      <td
        class="inf-table-cell"
      >
        На согласовании
      </td>
      <td
        class="inf-table-cell"
      />
    </tr>
  </tbody>
</table>
`;
